src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs

```diff
@@ -481,26 +520,134 @@ public async Task<ConnectorType> GetConnectorTypeAsync(NamedValue[] knownParamet
             }
         }
 
-        internal async Task<ConnectorType> GetConnectorTypeInternalAsync(NamedValue[] knownParameters, ConnectorType connectorType, BaseRuntimeConnectorContext runtimeContext, CancellationToken cancellationToken)
+        /// <summary>
+        /// Dynamic intellisense (dynamic property/schema) on a given connector type (coming from a parameter or returned from GetConnectorTypeAsync).
+        /// </summary>
+        /// <param name="knownParameters">Known parameters.</param>
+        /// <param name="connectorType">Connector type for which we need the (dynamic) type.</param>
+        /// <param name="runtimeContext">Connector context.</param>
+        /// <param name="maxCalls">Max number of recursive network calls.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>Formula Type determined by dynamic Intellisense.</returns>
+        public async Task<ConnectorType> GetConnectorTypeAsync(NamedValue[] knownParameters, ConnectorType connectorType, BaseRuntimeConnectorContext runtimeContext, int maxCalls, CancellationToken cancellationToken)
+        {
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                runtimeContext.ExecutionLogger?.LogInformation($"Entering in {this.LogFunction(nameof(GetConnectorTypeAsync))}, with {LogKnownParameters(knownParameters)} and maxCalls {maxCalls} for {LogConnectorType(connectorType)}");
+                (ConnectorType connectorType2, maxCalls) = await GetConnectorTypeInternalAsync(knownParameters, connectorType, runtimeContext, maxCalls, cancellationToken).ConfigureAwait(false);
+                runtimeContext.ExecutionLogger?.LogInformation($"Exiting {this.LogFunction(nameof(GetConnectorTypeAsync))}, returning from {nameof(GetConnectorTypeInternalAsync)} with {LogConnectorType(connectorType2)}, maxCalls {maxCalls}");
+                return connectorType2;
+            }
+            catch (Exception ex)
+            {
+                maxCalls--;
+                runtimeContext.ExecutionLogger?.LogException(ex, $"Exception in {this.LogFunction(nameof(GetConnectorTypeAsync))}, Context {LogKnownParameters(knownParameters)}, MaxCalls {maxCalls} {LogConnectorType(connectorType)}, {LogException(ex)}");
```

Comment:
> MaxCalls {maxCalls [](http://example.com/codeflow?start=177&length=19)

If MaxCalls was 200, and then we threw after making 50 calls, it will still be set to 200. And then this error will say 199 calls.